src/error.rs

```diff
@@ -144,18 +144,17 @@ where
 #[derive(Error, Debug, PartialEq, Eq)]
 #[non_exhaustive]
 pub enum SqlErr {
-    /// error for inserting a record with a key that already exists in the table
-    #[error("Cannot have record with same key")]
-    UniqueConstraintViolation(),
-    /// error for Foreign key is not primary key
-    #[error("Cannot add non-primary key from other table")]
-    ForeignKeyConstraintViolation(),
+    /// error for duplicate record in unique field or primary key field
+    #[error("Unique Constraint Violated")]
+    UniqueConstraintViolation,
```

Comment:
My Question is why remove the `()`? It is a placeholder, in case some day, we'd like to add some content.

Reply:
Actually, I though we shipped this enum somewhere already.
But then can we actually shove a little more context inside them?